services/registry/handlers/maintainers.js

```diff
@@ -16,88 +10,70 @@ module.exports = [
   )
 ];
 
 async function maintainers(context, { namespace, host, name }) {
-  const pkg = await Package.objects
-    .get({
-      active: true,
+  const [err, maintainers] = await context.storageApi
+    .listPackageMaintainers({
+      namespace,
+      host,
       name,
-      'namespace.active': true,
-      'namespace.name': namespace
+      bearer: context.user ? context.user.name : null,
+      page: Number(context.url.query.page) || 0,
+      status: context.url.query.status
     })
-    .catch(Package.objects.NotFound, () => null);
+    .then(xs => [null, xs], xs => [xs, null]);
 
-  if (!pkg) {
+  if (err) {
+    if (err.status === 404) {
+      return response.error(
+        `"${namespace}@${host}/${name}" does not exist.`,
+        404
+      );
+    }
+
+    context.logger.error(err.message || err);
     return response.error(
-      `"${namespace}@${host}/${name}" does not exist.`,
-      404
+      `Caught error fetching maintainers for "${namespace}@${host}/${name}".`,
+      500
     );
   }
-
-  const namespaces = await Namespace.objects
-    .filter({
-      'maintainers.package_id': pkg.id,
-      'maintainers.accepted': true,
-      'maintainers.active': true
-    })
-    .then();
-
-  const objects = namespaces.map(ns => ns.name).sort();
-  return response.json({ objects });
+  return response.json(maintainers);
 }
 
 // Invite a maintainer. Maintainers are namespaces, which might be a single user or a group of users.
 // More correctly: maintainership is a relationship between a namespace and a package.
 async function invite(context, { namespace, host, name, invitee }) {
-  if (!context.pkg) {
-    return response.error(`"${namespace}@${host}/${name}" not found.`, 404);
-  }
-
-  if (!context.invitee) {
-    return response.error(`${invitee} not found.`, 404);
-  }
-
-  const existing = await Maintainer.objects
-    .get({
-      namespace: context.invitee,
-      package: context.pkg
+  const [err, invite] = await context.storageApi
+    .invitePackageMaintainer({
+      namespace,
+      host,
+      name,
+      bearer: context.user.name,
+      to: invitee
     })
-    .catch(Maintainer.objects.NotFound, () => null);
-  if (existing) {
-    if (existing.active === false) {
-      return response.message(
-        `${invitee} has already declined to maintain ${namespace}@${host}/${name}.`
-      );
-    }
-    if (existing.accepted === false) {
-      return response.message(
-        `${invitee} has already been invited to maintain ${namespace}@${host}/${name}.`
-      );
-    }
+    .then(xs => [null, xs], xs => [xs, null]);
+
+  if (err) {
+    const msg = {
+      'maintainer.invite.invitee_dne': `Unknown namespace: "${invitee}".`,
+      'maintainer.invite.package_dne': `Unknown package: "${invitee}".`,
+      'maintainer.invite.already_accepted': `Namespace "${invitee}" is already a member.`,
+      'maintainer.invite.already_declined': `Namespace "${invitee}" has declined this invite.`
+    }[err.code];
```

Comment:
This is (more or less) what I'm thinking error enumeration should look like, per those many TODO comments.